### ISSUE TITLE
Incorrect Nested Message in Javascript

### ISSUE THREAD

**Issue by pradyuman** [google/protobuf#2134]
When referring to a nested message in a different file, you write something like this:
```
import "file"

message Test {
  File.Type fileType = 1;
}
```

```
// "file"
message File {
  message Type {
  }
}
```

However, this serializes to:
```
var value = new file_pb.Type;
// should be:
// var value = new file_pb.File.Type;
```

**Comment by pradyuman** [comment]
I think this is more of a bug than a question? Or will this be marked as a question until it is confirmed as a bug?

**Comment by xfxyjwf** [comment]
Yeah, I'm not that familiar with javascript and am waiting on someone who is more familiar with it to follow up. Given that nested messages are very common use cases, it seems unlikely the javascript would have issues supporting it.

**Comment by acozzette** [comment]
I gave this a try and it seemed to work as it should.

file1.proto:
```
syntax = "proto3";

package acozzette;

import "file2.proto";

message Test {
  File.Type file_type = 1;
}
```

file2.proto:
```
syntax = "proto3";

package acozzette;

message File {
  message Type {
  }
}
```

`./src/protoc --js_out=. file1.proto file2.proto`

Then inside the generated `file.js` I see:
```
...
goog.provide('proto.acozzette.File');
goog.provide('proto.acozzette.File.Type');
...
```

Pradyuman, does this behavior match what you were seeing?

**Comment by pradyuman** [comment]
My generated protos don't include `goog.provide`. In the equivalent of the `file2_pb.js`, I have:
```
goog.exportSymbol('proto.packagename.File');
goog.exportSymbol('proto.packagename.File.Type');
```

However, in the equivalent of `file1_pb.js`, I have:
```
var value = new packagename_file1_pb.Type
// should be
var value = new packagename_file1_pb.File.Type
```

I'm using protoc v3.0.0. I will try upgrading to v3.1.0 and see what happens.

**Comment by pradyuman** [comment]
Just upgraded to protoc v3.1.0 - same issue.

**Comment by acozzette** [comment]
Thanks, Pradyuman, I see how to reproduce the problem now. It looks to me like it's only a problem if you're using CommonJS-style imports. I'll take a closer look and see if I can fix the issue.

**Comment by pradyuman** [comment]
Thanks! Also, is there a plan to update google-protobuf on npm to be compatible with protobuf 3.1? I updated protobuf but then realized that it's not yet compatible with the version on npm (`TypeError: jspb.Message.getFieldWithDefault is not a function`).

**Comment by acozzette** [comment]
I have a pull request with a fix here: https://github.com/google/protobuf/pull/2193

@TeBoring, do you know when the npm package will be updated?

**Comment by pradyuman** [comment]
Thanks!

**Comment by acozzette** [comment]
This is now fixed on the 3.1.x branch.
